var/spack/repos/builtin/packages/py-libclang/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyLibclang(PythonPackage):
+    """The repository contains code that taken from the LLVM project, to make
+    it easier to install clang's python bindings."""
+
+    homepage = "https://github.com/sighingnow/libclang"
+
+    url = "https://github.com/sighingnow/libclang/archive/refs/tags/llvm-11.1.0.tar.gz"
+
+    version(
+        "11.1.0",
+        sha256="0b53b3c237725e193c4d2bbbe096f1a1da0f0e5cd528f2892e4dfed3c8fe9506",
+    )
+    version(
+        "11.0.1",
+        sha256="739ae984a4a4043ae4d3b4db74597a36a8e46b6f0cbd139c7d2703faf40c5390",
+    )
+    version(
+        "11.0.0",
+        sha256="aec204414ea412e4d4e041b0bf48123881338ac723bbcfa948f2a1b92a2428b5",
+    )
+    version(
+        "10.0.1",
+        sha256="c15d8f97c4d0f3d4501e8b2625b343569fd92690afebe6260a2c64463d713995",
+    )
+    version(
+        "9.0.1",
+        sha256="fc84e7bf3b0eb4f11c496d6603f111e3d8cda97094d6c9c512361371f1b76f1c",
+    )
+
+    depends_on("python@2.7:2.8,3.3:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-wheel", type=("build", "run"))
+
+    for ver in ["9", "10", "11"]:
+        depends_on("llvm@" + ver, when="@" + ver, type="build")
+
+    def patch(self):
+        filter_file(
+            r"source_dir = './native/'",
+            r"source_dir = '{0}'".format(self.spec["llvm"].prefix.lib),
```

Comment:
Should this be `spec['llvm'].libs.directories[0]`? Not all OSes use `prefix.lib`, some use `prefix.lib64`

Reply:
That would be better but I get the following error:
```console
==> Error: NoLibrariesError: Unable to recursively locate llvm libraries in /opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-10.3.0/llvm-11.1.0-dvkyuwd3s3524vkpkrme56zf6aft55ud
```

Reply:
Just add this to the LLVM package:
```python
@property
def libs(self):
    shared = '+shared_libs' in self.spec
    return find_libraries('libc++', self.prefix, shared=shared, recursive=True)
```
or something like that.

Reply:
I think this has to with the fact that there is no library named, `libllvm.so` in the llvm lib directory.

Reply:
Correct, if the library name is different than the package name, you need to add a custom `libs` attribute. `'libc++'` might not actually be the best name since I think that is only available if you build with `+libcxx`. Are there any libraries that always get installed?